cmake/Intel.cmake

```diff
@@ -17,10 +17,15 @@ if(DEBUG)
     endif()
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -ftrapuv")
 else()
-    set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -debug minimal -qoverride-limits")
-    set(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE} -fp-model consistent")
-    set(CMAKE_C_FLAGS_RELEASE       "-O2 -debug minimal")
-    set(FAST "-fast-transcendentals")
+    if(SUPAFAST)
+      set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -fp-model precise -assume buffered_stdout -fno-alias -align all -debug minimal -qoverride-limits -ftz")
+      set(CMAKE_C_FLAGS_RELEASE       "-O3 -fp-model precise -debug minimal -qoverride-limits -ftz")
+    else()
+      set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -debug minimal -qoverride-limits")
+      set(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE} -fp-model consistent")
+      set(CMAKE_C_FLAGS_RELEASE       "-O2 -debug minimal")
+      set(FAST "-fast-transcendentals")
+    endif()
```

Comment:
@SamuelTrahanNOAA, @junwang-noaa, @DusanJovic-NOAA, For this SUPAFAST build option, I was wondering if we can/need rename it into PRODUCTIOM (or PROD). Currently, I believe we have DEBUG, RELEASE options. Another thing related, should we also test this SUPAFAST/PRODUCTION/PROD option for all ufs-weather-model regression tests? Especially to test if the decomp/restart tests can pass the regression tests.

Reply:
I think it should replace "FAST" but not until it has been more widely tested.

Reply:
I would like comment from others about a suggested name for this option before I change it.

Reply:
@SamuelTrahanNOAA, yeah, FAST is another good naming option. Would like to hear opinions from others (@junwang-noaa, @DusanJovic-NOAA, and other code managers) on these naming options: FAST/SUPAFAST/PRODUCTION/PROD, etc. Thanks!

Reply:
There is already a "FAST" option, so we would be replacing something that other groups might already be using. Naming it "PROD" or "PRODUCTION" wouldn't be great, since the utility of the optimization options are not limited to production.

Reply:
Some time ago we decided to support only two build options, 'Debug' and "Release'. No special options like FAST/PROD/PRODUCTION etc. If the compiler options added or changed by this PR are bit-by-bit reproducible then we should use them in all test ie. they should be enabled by default whenever 'Release' build type is selected. If they are not reproducible they shouldn't be used in any test/configuration. Similar for 'Debug' build type. Debug build type is used for debugging, Release for all other use cases (including production) and it should be as fast as possible. I do not see the need for any other build option.

Reply:
You need `-fp-model consistent` on heterogenous clusters like Jet if you want the different partitions get the same result. You can't afford that luxury in operations. That means we do need three options:

1. Debugging. (The current "Debug" target.)
2. Support for heterogenous clusters. (The current "Release" target.)
3. Runs at a usable speed. (What this PR calls "SUPAFAST.")

Reply:
We always run tests on the same Jet cluster (xjet), if -fp-model precise is reproducible (is it?), then I do not think we need two different 'Release' versions, we should just use one.

Reply:
Unfortunately, the FV3_HRRR suite cap cannot compile in 8 hours. Until we have a way to set a different optimization option for the caps, this can't be used for all suites.

Does anyone know why `-fp-model consistent` was in the build options to begin with? Surely someone put it there for a reason?

Reply:
@climbfuji Do you remember when and why we started to use -fp-model consistent?